src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = Object.assign({}, state);
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete currentState[property];
+        }
+        break;
+
+      case 'addProperties':
+        for (const entry of Object.entries(action.extraData)) {
+          currentState[entry[0]] = entry[1];
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(currentState)) {
+          delete currentState[key];
+        }
```

Comment:
Instead of deleting properties one by one, you can simply assign an empty object to 'currentState' when the action type is 'clear'. This is more efficient and aligns with the action's intent.